### DISCUSSION TITLE
多动态IP时，优化ip的确定

### DISCUSSION THREAD

**Discussion by vnwnv** [NewFuture/DDNS#249] [Ideas]
**Is your feature request related to a problem? Please describe.**
多动态ip场景下，DNS解析变来变去。
**Describe the solution you'd like**
支持手动/自动选择ip，在多播、链路聚合等多动态ip场景下的设备正常工作
**Describe alternatives you've considered**
1、同一时间多次（次数可以用户指定，与负载比例相关）向指定的index4（如：ip.3322.net）请求ip，记录所有结果，并选择其中一个ip。下次获取ip后比较两次结果，如果选择的ip没有更换，则不改变dns解析。如果ip均发生改变，可根据ip的ISP、位置等信息自动选择，若都相同则对应多播情形，可以任选其一。
2、编译新版本，适用于路由器系统，支持通过curl指定端口获取ip地址，然后通过端口转发/DMZ到达指定的主机。
**Additional context**
临时解决方案：在路由器中设置静态路由规则，让运行ddns设备的所有出入流量都经过指定的端口

**Comment by ZirrorDmage** [comment]
我思考了一下这样设计所对应的场景，是同时具有两条入户宽带，且都具有公网IP，通过mwan3等进行链路聚合和负载均衡。
首先这个场景在实际情况下并不那么常见，且对于运行于路由器内网下的ddns设备，想通过自身查询到上级公网IP对应的不同ISP是存在困难的，且对比二者的意义不大，何不考虑用两个二级域名进行解析？
其次，此项目基于Python运行，可以在多平台上无障碍运行，例如我当前使用时，将该代码上传至运行了openwrt的路由器内，安装Python3，运行该源码不存在缺少依赖的问题。如果你同我一样将代码运行于路由端，则可以通过获取指定网卡的方式获得不同的公网IP，也能够解决你的需求。
